src/feat/wave-reader.h

```diff
@@ -171,6 +172,11 @@ class WaveHolder {
     t_.Swap(&(other->t_));
   }
 
+  bool ExtractRange(const WaveHolder &other, const std::string &range) {
+    KALDI_ERR << "ExtractRange is not defined for this type of holder.";
```

Comment:
Up to you, not trying to convince you at all; I just want to comment on the "against the grain" part. C++11 makes such constructs easier to implement, with library metafunctions like std::enable_if (many borrowed from boost, naturally).

Having just written a massively multithreaded service with non-homogenous interacting threads in pure C++11, I would say that it feels different from the C++ as we knew it before, to the point where it feels like a wholly different language. Personally I liked it, as it certainly makes leaking memory harder and its library multithreading is piece of cake, but not everybody is so amazed. But all in all it seems that the "grain" is shifting its direction these days, as fully compatible C++ compilers become widely available.